### ISSUE TITLE
How do I make sure I'm using adam

### ISSUE THREAD

**Issue by menglin0320** [BVLC/caffe#4176]
I tried to use adam and my .sh file looks is shown below

# The train/test net protocol buffer definition
# this follows "ADAM: A METHOD FOR STOCHASTIC OPTIMIZATION"
net: "menglin_try/cardata_clean.prototxt"
# test_iter specifies how many forward passes the test should carry out.
# In the case of MNIST, we have test batch size 100 and 100 test iterations,
# covering the full 10,000 testing images.
test_iter: 100
# Carry out testing every 500 training iterations.
test_interval: 500
# All parameters are from the cited paper above
base_lr: 0.001
momentum: 0.9
momentum2: 0.999
# since Adam dynamically changes the learning rate, we set the base learning
# rate to a fixed value
lr_policy: "fixed"
# Display every 100 iterations
display: 100
# The maximum number of iterations
max_iter: 10000
# snapshot intermediate results
snapshot: 5000
snapshot_prefix: "menglin_try/snapshot"
# solver mode: CPU or GPU
type: "Adam"
solver_mode: GPU

But I see I0519 01:11:09.896891  3318 sgd_solver.cpp:106] Iteration 3400, lr = 0.001
on training. 
Should I see adam_solver.cpp:"a line number" if everything is correct?

**Comment by seanbell** [comment]
If you want to convince yourself it's correct, you can add a print statement inside `adam_solver.cpp`.  Notice that ADAM inherits from the SGD solver, so you should expect to see the print statement come from `sgd_solver.cpp`.

In general, please ask future usage questions on the mailing list.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by menglin0320** [comment]
Ok, thanks
